sycl/include/CL/sycl/handler.hpp

```diff
@@ -392,6 +392,19 @@ class handler {
         std::move(CommandGroup), std::move(MQueue));
 
     EventRet = detail::createSyclObjFromImpl<event>(Event);
+
+    // Waiting for copy command to complete here as SYCL specification says that
+    // the SYCL runtime must ensure that data is copied to the destination once
+    // the command group has completed execution.
+    switch (MCGType) {
+    case detail::CG::COPY_ACC_TO_PTR:
+    case detail::CG::COPY_PTR_TO_ACC:
+    case detail::CG::COPY_ACC_TO_ACC:
```

Comment:
Is this also true for accessor to accessor?
It seems pessimistic to me, in the sense there is no difference compared with a normal kernel implementing some transfer between 2 buffers...

Reply:
Agree with you, I think even copying accessor to ptr and vice versa should not be blocking as well.

But currently Specification says:

> The most recent copy of the memory object may reside on any context controlled by the SYCL runtime, or on the host in a pointer controlled by the SYCL runtime. **The SYCL runtime will ensure that data is copied to the destination once the command group has completed execution.**

There is already written code that relies on that.